tests/test_frame.py

```diff
@@ -4688,3 +4688,112 @@ def test_unstack() -> None:
         ),
         pd.DataFrame,
     )
+
+
+def test_from_records() -> None:
+    
+    #test with np.ndarray
+    arr = np.array([[1, "a"], [2, "b"]], dtype=object)
```

Comment:
```suggestion
    arr = np.array([[1, "a"], [2, "b"]], dtype=object).reshape(2, 2)
```
CI failing here - has to do with differences of python 3.10 and later versions in terms of numpy support.